
<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN"
  "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">

<html xmlns="http://www.w3.org/1999/xhtml">
  <head>
    <meta http-equiv="X-UA-Compatible" content="IE=Edge" />
    <meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
    <title>Output specification for atac.wdl &#8212; ENCODE genomic pipelines  documentation</title>
    <link rel="stylesheet" href="_static/alabaster.css" type="text/css" />
    <link rel="stylesheet" href="_static/pygments.css" type="text/css" />
    <script type="text/javascript" src="_static/documentation_options.js"></script>
    <script type="text/javascript" src="_static/jquery.js"></script>
    <script type="text/javascript" src="_static/underscore.js"></script>
    <script type="text/javascript" src="_static/doctools.js"></script>
    <link rel="index" title="Index" href="genindex.html" />
    <link rel="search" title="Search" href="search.html" />
    <link rel="prev" title="Output specification for chip.wdl" href="output_chip.html" />
   
  <link rel="stylesheet" href="_static/custom.css" type="text/css" />
  
  
  <meta name="viewport" content="width=device-width, initial-scale=0.9, maximum-scale=0.9" />

  </head><body>
  

    <div class="document">
      <div class="documentwrapper">
        <div class="bodywrapper">
          <div class="body" role="main">
            
  <div class="section" id="output-specification-for-atac-wdl">
<h1>Output specification for <code class="docutils literal notranslate"><span class="pre">atac.wdl</span></code><a class="headerlink" href="#output-specification-for-atac-wdl" title="Permalink to this headline">Â¶</a></h1>
<div class="toctree-wrapper compound">
</div>
<p>All output filenames keep prefixes from corresponding input filenames. For example. If you have started from <code class="docutils literal notranslate"><span class="pre">REP1.fastq.gz</span></code> and <code class="docutils literal notranslate"><span class="pre">REP2.fastq.gz</span></code> then corresponding alignment log for each replicate has a filename of <code class="docutils literal notranslate"><span class="pre">REP1.flagstat.qc</span></code> and <code class="docutils literal notranslate"><span class="pre">REP2.flagstat.qc</span></code>, respectively.</p>
<p>Some summarizing QC files do not have any prefix. Find <code class="docutils literal notranslate"><span class="pre">qc.json</span></code> and <code class="docutils literal notranslate"><span class="pre">qc.html</span></code> for final QC and HTML report.</p>
<ol class="arabic simple">
<li><code class="docutils literal notranslate"><span class="pre">DNANexus</span></code>: If you choose to use <code class="docutils literal notranslate"><span class="pre">dxWDL</span></code> and run pipelines on DNANexus platform, then output will be stored on the specified output directory without any subdirectories.</li>
<li><code class="docutils literal notranslate"><span class="pre">Cromwell</span></code>: Otherwise <code class="docutils literal notranslate"><span class="pre">Cromwell</span></code> will store outputs for each task under <code class="docutils literal notranslate"><span class="pre">cromwell-executions/[WORKFLOW_ID]/call-[TASK_NAME]/shard-[IDX]</span></code>. For all tasks except <code class="docutils literal notranslate"><span class="pre">idr</span></code> and <code class="docutils literal notranslate"><span class="pre">overlap</span></code>, <code class="docutils literal notranslate"><span class="pre">[IDX]</span></code> means a zero-based index for each replicate but for tasks <code class="docutils literal notranslate"><span class="pre">idr</span></code> and <code class="docutils literal notranslate"><span class="pre">overlap</span></code> it stands for a zero-based index for all possible pair of replicates. For example, you have 3 replicates and all possible combination of two replicates are <code class="docutils literal notranslate"><span class="pre">[(rep1,rep2),</span> <span class="pre">(rep1,rep3),</span> <span class="pre">(rep2,rep3)]</span></code>. Therefore, <code class="docutils literal notranslate"><span class="pre">call-idr/shard-2</span></code> should be an output directory for the pair of replicate 2 and 3.</li>
</ol>
<p>For more details, refer to the file table section in an HTML report generated by the pipeline. Files marked as (E) are outputs to be uploaded during ENCODE accession.</p>
<table border="1" class="docutils">
<colgroup>
<col width="33%" />
<col width="33%" />
<col width="33%" />
</colgroup>
<thead valign="bottom">
<tr class="row-odd"><th class="head">Task name</th>
<th class="head">File</th>
<th class="head">Description</th>
</tr>
</thead>
<tbody valign="top">
<tr class="row-even"><td>trim_adapter</td>
<td>*.trim.fastq.gz</td>
<td>adapter-trimmed FASTQ</td>
</tr>
<tr class="row-odd"><td>trim_adapter</td>
<td>merge_fastqs_R?_*.fastq.gz</td>
<td>Merged and adapter-trimmed FASTQ</td>
</tr>
<tr class="row-even"><td>bowtie2</td>
<td>*.bam</td>
<td>Raw BAM</td>
</tr>
<tr class="row-odd"><td>bowtie2</td>
<td>*.bai</td>
<td>BAI for Raw BAM</td>
</tr>
<tr class="row-even"><td>bowtie2</td>
<td>*.align.log</td>
<td>Bowtie2 log for mapping</td>
</tr>
<tr class="row-odd"><td>bowtie2</td>
<td>*.flagstat.qc</td>
<td>Samtools flagstat log for raw BAM</td>
</tr>
<tr class="row-even"><td>filter</td>
<td>*.nodup.bam</td>
<td>Filtered/deduped BAM</td>
</tr>
<tr class="row-odd"><td>filter</td>
<td>*.nodup.flagstat.qc</td>
<td>Samtools flagstat log for filtered/deduped BAM</td>
</tr>
<tr class="row-even"><td>filter</td>
<td>*.dup.qc</td>
<td>Picard/sambamba markdup log</td>
</tr>
<tr class="row-odd"><td>filter</td>
<td>*.pbc.qc</td>
<td>PBC QC log</td>
</tr>
<tr class="row-even"><td>bam2ta</td>
<td>*.tagAlign.gz</td>
<td>TAG-ALIGN generated from filtered BAM</td>
</tr>
<tr class="row-odd"><td>bam2ta</td>
<td>*.N.tagAlign.gz</td>
<td>Subsampled (N reads) TAG-ALIGN generated from filtered BAM</td>
</tr>
<tr class="row-even"><td>bam2ta</td>
<td>*.tn5.tagAlign.gz</td>
<td>TN5-shifted TAG-ALIGN</td>
</tr>
<tr class="row-odd"><td>spr</td>
<td>*.pr1.tagAlign.gz</td>
<td>1st pseudo-replicated TAG-ALIGN</td>
</tr>
<tr class="row-even"><td>spr</td>
<td>*.pr2.tagAlign.gz</td>
<td>2nd pseudo-replicated TAG-ALIGN</td>
</tr>
<tr class="row-odd"><td>pool_ta</td>
<td>*.tagAlign.gz</td>
<td>Pooled TAG-ALIGN from all replciates</td>
</tr>
<tr class="row-even"><td>xcor</td>
<td>*.cc.plot.pdf</td>
<td>Cross-correlation plot PDF</td>
</tr>
<tr class="row-odd"><td>xcor</td>
<td>*.cc.plot.png</td>
<td>Cross-correlation plot PNG</td>
</tr>
<tr class="row-even"><td>xcor</td>
<td>*.cc.qc</td>
<td>Cross-correlation analysis score log</td>
</tr>
<tr class="row-odd"><td>xcor</td>
<td>*.cc.fraglen.txt</td>
<td>Estimated fragment length</td>
</tr>
<tr class="row-even"><td>macs2</td>
<td>*.narrowPeak.gz</td>
<td>NARROWPEAK</td>
</tr>
<tr class="row-odd"><td>macs2</td>
<td>*.bfilt.narrowPeak.gz</td>
<td>Blacklist-filtered NARROWPEAK</td>
</tr>
<tr class="row-even"><td>macs2</td>
<td>*.pval.signal.bigwig</td>
<td>p-val signal BIGWIG</td>
</tr>
<tr class="row-odd"><td>macs2</td>
<td>*.fc.signal.bigwig</td>
<td>fold enrichment signal BIGWIG</td>
</tr>
<tr class="row-even"><td>macs2</td>
<td>*.frip.qc</td>
<td>Fraction of read (TAG-ALIGN) in peaks (NARROWPEAK)</td>
</tr>
<tr class="row-odd"><td>idr</td>
<td>*.*Peak.gz</td>
<td>IDR NARROWPEAK</td>
</tr>
<tr class="row-even"><td>idr</td>
<td>*.bfilt.*Peak.gz</td>
<td>Blacklist-filtered IDR NARROWPEAK</td>
</tr>
<tr class="row-odd"><td>idr</td>
<td>*.txt.png</td>
<td>IDR plot PNG</td>
</tr>
<tr class="row-even"><td>idr</td>
<td>*.txt.gz</td>
<td>Unthresholded IDR output</td>
</tr>
<tr class="row-odd"><td>idr</td>
<td>*.log</td>
<td>IDR STDOUT log</td>
</tr>
<tr class="row-even"><td>idr</td>
<td>*.frip.qc</td>
<td>Fraction of read (TAG-ALIGN) in peaks (IDR NARROWPEAK)</td>
</tr>
<tr class="row-odd"><td>overlap</td>
<td>*.*Peak.gz</td>
<td>Overlapping NARROWPEAK</td>
</tr>
<tr class="row-even"><td>overlap</td>
<td>*.bfilt.*Peak.gz</td>
<td>Blacklist-filtered overlapping NARROWPEAK</td>
</tr>
<tr class="row-odd"><td>overlap</td>
<td>*.frip.qc</td>
<td>Fraction of read (TAG-ALIGN) in peaks (overlapping NARROWPEAK)</td>
</tr>
<tr class="row-even"><td>reproducibility</td>
<td>*.reproducibility.qc</td>
<td>Reproducibililty QC log</td>
</tr>
<tr class="row-odd"><td>qc_report</td>
<td>qc.html</td>
<td>Final HTML QC report</td>
</tr>
<tr class="row-even"><td>qc_report</td>
<td>qc.html</td>
<td>Final QC JSON</td>
</tr>
</tbody>
</table>
</div>


          </div>
        </div>
      </div>
      <div class="sphinxsidebar" role="navigation" aria-label="main navigation">
        <div class="sphinxsidebarwrapper"><div class="relations">
<h3>Related Topics</h3>
<ul>
  <li><a href="index.html">Documentation overview</a><ul>
      <li>Previous: <a href="output_chip.html" title="previous chapter">Output specification for <code class="docutils literal notranslate"><span class="pre">chip.wdl</span></code></a></li>
  </ul></li>
</ul>
</div>
  <div role="note" aria-label="source link">
    <h3>This Page</h3>
    <ul class="this-page-menu">
      <li><a href="_sources/output_atac.rst.txt"
            rel="nofollow">Show Source</a></li>
    </ul>
   </div>
<div id="searchbox" style="display: none" role="search">
  <h3>Quick search</h3>
    <div class="searchformwrapper">
    <form class="search" action="search.html" method="get">
      <input type="text" name="q" />
      <input type="submit" value="Go" />
      <input type="hidden" name="check_keywords" value="yes" />
      <input type="hidden" name="area" value="default" />
    </form>
    </div>
</div>
<script type="text/javascript">$('#searchbox').show(0);</script>
        </div>
      </div>
      <div class="clearer"></div>
    </div>
    <div class="footer">
      &copy;2018, ENCODE DCC.
      
      |
      Powered by <a href="http://sphinx-doc.org/">Sphinx 1.7.3</a>
      &amp; <a href="https://github.com/bitprophet/alabaster">Alabaster 0.7.10</a>
      
      |
      <a href="_sources/output_atac.rst.txt"
          rel="nofollow">Page source</a>
    </div>

    

    
  </body>
</html>